tutorials/docs-00-getting-started/index.qmd

```diff
@@ -16,96 +16,149 @@ Pkg.instantiate();
-m = Normal(0, 1)
+α, β = shape(s²), scale(s²)
+
+# We defined m ~ Normal(0, sqrt(s²)), and the normal-inverse gamma
+# distribution is defined by m ~ Normal(µ, sqrt(s²/λ)), which lets
+# us identify the following
+µ, λ = 0, 1
+
 data = [1.5, 2]
 x_bar = mean(data)
-N = length(data)
+n = length(data)
+v = var(data; corrected=false)
+
+µ´ = ((λ * µ) + (n * x_bar)) / (λ + n)
+λ´ = λ + n
+α´ = α + n / 2
+β´ = β + ((n * v) + ((λ * n * (x_bar - µ)^2) / (λ + n))) / 2
 
-mean_exp = (m.σ * m.μ + N * x_bar) / (m.σ + N)
+(µ´, λ´, α´, β´)
 ```
 
-We can also compute the updated variance
+We can use these to analytically calculate, for example, the expectation values of the mean and variance parameters in the posterior distribution: $E[m] = \mu'$, and $E[s^2] = \beta'/(\alpha
+ - 1)$.
 
 ```{julia}
-updated_alpha = shape(s²) + (N / 2)
-updated_beta =
-    scale(s²) +
-    (1 / 2) * sum((data[n] - x_bar)^2 for n in 1:N) +
-    (N * m.σ) / (N + m.σ) * ((x_bar)^2) / 2
-variance_exp = updated_beta / (updated_alpha - 1)
+mean_exp = µ´
+variance_exp = β´ / (α´ - 1)
+
+(mean_exp, variance_exp)
 ```
 
-Finally, we can check if these expectations align with our HMC approximations from earlier. We can compute samples from a normal-inverse gamma following the equations given [here](https://en.wikipedia.org/wiki/Normal-inverse-gamma_distribution#Generating_normal-inverse-gamma_random_variates).
+Alternatively, we can compare the two distributions visually (one parameter at a time).
+To do so, we will directly sample from the analytical posterior distribution using the procedure described in [Wikipedia](https://en.wikipedia.org/wiki/Normal-inverse-gamma_distribution#Generating_normal-inverse-gamma_random_variates).
 
 ```{julia}
-function sample_posterior(alpha, beta, mean, lambda, iterations)
+function sample_posterior_analytic(µ´, λ´, α´, β´, iterations)
     samples = []
     for i in 1:iterations
-        sample_variance = rand(InverseGamma(alpha, beta), 1)
-        sample_x = rand(Normal(mean, sqrt(sample_variance[1]) / lambda), 1)
-        samples = append!(samples, sample_x)
+        sampled_s² = rand(InverseGamma(α´, β'))
+        sampled_m = rand(Normal(µ´, sqrt(sampled_s² / λ´)))
+        samples = push!(samples, (sampled_m, sampled_s²))
     end
     return samples
 end
 
-analytical_samples = sample_posterior(updated_alpha, updated_beta, mean_exp, 2, 1000);
+analytical_samples = sample_posterior_analytic(µ´, λ´, α´, β´, 1000);
 ```
 
+Comparing the distribution of the mean:
+
 ```{julia}
-density(analytical_samples; label="Posterior (Analytical)")
+density([s[1] for s in analytical_samples]; label="Posterior (Analytical)")
 density!(chn[:m]; label="Posterior (HMC)")
 ```
+
+and the distribution of the variance:
+
+```{julia}
+density([s[2] for s in analytical_samples]; label="Posterior (Analytical)")
+density!(chn[:s²]; label="Posterior (HMC)")
+```
+
+we see that our HMC sampler has given us a good representation of the posterior distribution.
```

Comment:
In fact, both the HMC sample as well as our custom sampler are affected by randomness and any time `quarto render` is run, there's no guarantee as to how well the distributions will line up.

This would of course be improved with more samples, but then the docs take longer to run.

Would it make sense to set the RNG seed where necessary?